packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -196,6 +196,28 @@ function BlockSelectionButton( { clientId, rootClientId, blockElement } ) {
 				if ( navigateDown ) {
 					nextTabbable = focus.tabbable.findNext( blockElement );
```

Comment:
I've been looking at this PR again and for me the root cause is in this line basically what we're saying here is:

If there's no next block, focus the next tabbable element after the current block element, the problem is that this doesn't exclude the content of the current block (because it comes after blockElement), so I think a proper if is just to continue calling "findNext" until we're out of the current block.

Something like this maybe?

```
nextTabbable = blockElement;
					do {
						nextTabbable = focus.tabbable.findNext( nextTabbable );
					} while ( blockElement.contains( nextTabbable ) );

```

Reply:
I am glad you knew what I was trying to do. 👍 This works perfectly. 🙇‍♂️ Thanks for suggestion.